R/imageutils.R

```diff
@@ -11,7 +11,27 @@ startPNG <- function(filename, width, height, res, ...) {
     grDevices::png
   }
 
-  args <- rlang::list2(filename=filename, width=width, height=height, res=res, ...)
+  # It's possible for width/height to be NULL (e.g., when using
+  # suspendWhenHidden=F w/ tabsetPanel()), which will lead to an error when
+  # attempting to open the device (rstudio/shiny#1409). In this case, ragg will
+  # actually segfault (instead of error), so explicitly throw an error before
+  # opening the device (r-lib/ragg#116, rstudio/shiny#3704)
+  check_empty_png_size <- function(x) {
+    if (length(x) > 0) return(x)
```

Comment:
Do you want to check for `x == 0` as well? Or do we not care what ragg does in that case

Reply:
ragg won't seqfault in that case (and I'm pretty sure they'll all, at least eventually, result in `Error  : figure margins too large`, so it didn't seem worth including that case